scripts/travis_gdal_install.sh

```diff
@@ -51,153 +51,156 @@ GDAL_CMAKE_OPTS=" -DBUILD_APPS=OFF \
-        -DGDAL_ENABLE_DRIVER_STACIT=OFF \ 
-        -DGDAL_ENABLE_DRIVER_STACTA=OFF \ 
-        -DGDAL_ENABLE_DRIVER_TERRAGEN=OFF \ 
-        -DGDAL_ENABLE_DRIVER_TGA=OFF \ 
-        -DGDAL_ENABLE_DRIVER_TIL=OFF \ 
-        -DGDAL_ENABLE_DRIVER_TSX=OFF \ 
-        -DGDAL_ENABLE_DRIVER_USGSDEM=OFF \  
-        -DGDAL_ENABLE_DRIVER_VRT=OFF \ 
-        -DGDAL_ENABLE_DRIVER_WCS=OFF \ 
-        -DGDAL_ENABLE_DRIVER_WEBP=OFF \ 
-        -DGDAL_ENABLE_DRIVER_WMS=OFF \ 
-        -DGDAL_ENABLE_DRIVER_WMTS=OFF \ 
-        -DGDAL_ENABLE_DRIVER_XPM=OFF \ 
-        -DGDAL_ENABLE_DRIVER_XYZ=OFF \ 
-        -DGDAL_ENABLE_DRIVER_ZARR=OFF \ 
-        -DGDAL_ENABLE_DRIVER_ZMAP=OFF \ 
-        -DOGR_BUILD_OPTIONAL_DRIVERS=OFF \ 
-        -DOGR_ENABLE_DRIVER_AVC=OFF \ 
-        -DOGR_ENABLE_DRIVER_CAD=OFF \ 
-        -DOGR_ENABLE_DRIVER_CSV=ON \ 
-        -DOGR_ENABLE_DRIVER_CSW=OFF \ 
-        -DOGR_ENABLE_DRIVER_DGN=ON \ 
-        -DOGR_ENABLE_DRIVER_DXF=ON \ 
-        -DOGR_ENABLE_DRIVER_EDIGEO=OFF \ 
-        -DOGR_ENABLE_DRIVER_ELASTIC=OFF \ 
-        -DOGR_ENABLE_DRIVER_FLATGEOBUF=ON \ 
-        -DOGR_ENABLE_DRIVER_GEOCONCEPT=OFF \ 
-        -DOGR_ENABLE_DRIVER_GEOJSON=ON \ 
-        -DOGR_ENABLE_DRIVER_GEORSS=OFF \ 
-        -DOGR_ENABLE_DRIVER_GML=ON \ 
-        -DOGR_ENABLE_DRIVER_GMT=ON \ 
-        -DOGR_ENABLE_DRIVER_GPKG=ON \ 
-        -DOGR_ENABLE_DRIVER_GPSBABEL=OFF \ 
-        -DOGR_ENABLE_DRIVER_GPX=ON \ 
-        -DOGR_ENABLE_DRIVER_IDRISI=OFF \ 
-        -DOGR_ENABLE_DRIVER_ILI=OFF \ 
-        -DOGR_ENABLE_DRIVER_JML=OFF \ 
-        -DOGR_ENABLE_DRIVER_KML=OFF \ 
-        -DOGR_ENABLE_DRIVER_LIBKML=OFF \ 
-        -DOGR_ENABLE_DRIVER_LVBAG=OFF \ 
-        -DOGR_ENABLE_DRIVER_MAPML=OFF \ 
-        -DOGR_ENABLE_DRIVER_MEM=OFF \ 
-        -DOGR_ENABLE_DRIVER_MSSQLSPATIAL=OFF \ 
-        -DOGR_ENABLE_DRIVER_MYSQL=OFF \ 
-        -DOGR_ENABLE_DRIVER_NAS=OFF \ 
-        -DOGR_ENABLE_DRIVER_NGW=OFF \ 
-        -DOGR_ENABLE_DRIVER_NTF=OFF \ 
-        -DOGR_ENABLE_DRIVER_ODBC=OFF \ 
-        -DOGR_ENABLE_DRIVER_ODS=OFF \ 
-        -DOGR_ENABLE_DRIVER_OPENFILEGDB=ON \ 
-        -DOGR_ENABLE_DRIVER_OSM=OFF \ 
-        -DOGR_ENABLE_DRIVER_PGDUMP=OFF \ 
-        -DOGR_ENABLE_DRIVER_PGEO=OFF \ 
-        -DOGR_ENABLE_DRIVER_PLSCENES=OFF \ 
-        -DOGR_ENABLE_DRIVER_S57=ON \ 
-        -DOGR_ENABLE_DRIVER_SDTS=OFF \ 
-        -DOGR_ENABLE_DRIVER_SELAFIN=OFF \ 
-        -DOGR_ENABLE_DRIVER_SHAPE=ON \ 
-        -DOGR_ENABLE_DRIVER_SQLITE=ON \ 
-        -DOGR_ENABLE_DRIVER_SVG=OFF \ 
-        -DOGR_ENABLE_DRIVER_SXF=OFF \ 
-        -DOGR_ENABLE_DRIVER_TAB=ON \ 
-        -DOGR_ENABLE_DRIVER_TIGER=OFF \ 
-        -DOGR_ENABLE_DRIVER_VDV=OFF \ 
-        -DOGR_ENABLE_DRIVER_VFK=OFF \ 
-        -DOGR_ENABLE_DRIVER_VRT=ON \ 
-        -DOGR_ENABLE_DRIVER_WASP=OFF \ 
-        -DOGR_ENABLE_DRIVER_WFS=OFF \ 
-        -DOGR_ENABLE_DRIVER_XLS=OFF \ 
+        -DGDAL_USE_MSSQL_NCLI=OFF \
+        -DGDAL_USE_MSSQL_ODBC=OFF \
+        -DGDAL_USE_ODBC=OFF \
```

Comment:
Apparently the new GHA runner image added incomplete ODBC support and Cmake found it, but couldn't build it successfully. I feel like a lot of optional features should have to be explicitly asked for. The build breakage I'm fixing could happen again in other corners of GDAL.